repair/repair.cc

```diff
@@ -2104,6 +2101,32 @@ future<> repair_service::repair_tablets(repair_uniq_id rid, sstring keyspace_nam
             std::vector<gms::inet_address> nodes;
             auto master_shard_id = m.master_shard_id;
             auto range = m.range;
+            bool select = true;
+            if (!ranges_specified.empty()) {
+                select = false;
+                for (auto& r : ranges_specified) {
+                    // When the management tool, e.g., scylla manager, uses
+                    // ranges option to select which ranges to repair for a
+                    // tablet table to schedule repair work, it needs to
+                    // disable tablet migration to make sure the node and token
+                    // range mapping does not change. Here, we check if the
+                    // user specified range and the range owned by the node
+                    // matches exactly, ignoring the case the two ranges
+                    // match partially due to tablet migration.
+                    if (range == r) {
+                        rlogger.debug("repair[{}] Select tablet table={}.{} tablet_id={} range={} replicas={} primary_replica_only={} ranges_specified={}",
+                            rid.uuid(), keyspace_name, table_name, m.id, m.range, m.replicas, primary_replica_only, ranges_specified);
+                        select = true;
+                        break;
+                    }
+                    co_await coroutine::maybe_yield();
+                }
+            }
```

Comment:
What about updating the repair history table? Will it update only the repaired tablets? Let's add a test for it.

Reply:
We already have the repair history table update test. The test added in this patch writes to repair history table too.